ros2_control_demo_hardware/include/ros2_control_demo_hardware/rrbot_transmissions_system_position_only.hpp

```diff
@@ -0,0 +1,96 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef ROS2_CONTROL_DEMO_HARDWARE__RRBOT_TRANSMISSIONS_SYSTEM_POSITION_ONLY_HPP_
+#define ROS2_CONTROL_DEMO_HARDWARE__RRBOT_TRANSMISSIONS_SYSTEM_POSITION_ONLY_HPP_
+
+#include <map>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "hardware_interface/handle.hpp"
+#include "hardware_interface/hardware_info.hpp"
+#include "hardware_interface/system_interface.hpp"
+#include "hardware_interface/types/hardware_interface_return_values.hpp"
+#include "rclcpp/clock.hpp"
+#include "rclcpp/logger.hpp"
+#include "rclcpp/macros.hpp"
+#include "rclcpp_lifecycle/state.hpp"
+#include "ros2_control_demo_hardware/visibility_control.h"
+#include "transmission_interface/transmission.hpp"
+
+namespace ros2_control_demo_hardware
+{
+class RRBotTransmissionsSystemPositionOnlyHardware : public hardware_interface::SystemInterface
+{
+public:
+  RCLCPP_SHARED_PTR_DEFINITIONS(RRBotTransmissionsSystemPositionOnlyHardware);
+
+  ROS2_CONTROL_DEMO_HARDWARE_PUBLIC
+  hardware_interface::CallbackReturn on_init(
+    const hardware_interface::HardwareInfo & info) override;
+
+  ROS2_CONTROL_DEMO_HARDWARE_PUBLIC
+  hardware_interface::CallbackReturn on_configure(
+    const rclcpp_lifecycle::State & previous_state) override;
+
+  ROS2_CONTROL_DEMO_HARDWARE_PUBLIC
+  std::vector<hardware_interface::StateInterface> export_state_interfaces() override;
+
+  ROS2_CONTROL_DEMO_HARDWARE_PUBLIC
+  std::vector<hardware_interface::CommandInterface> export_command_interfaces() override;
+
+  ROS2_CONTROL_DEMO_HARDWARE_PUBLIC
+  hardware_interface::CallbackReturn on_activate(
+    const rclcpp_lifecycle::State & previous_state) override;
+
+  ROS2_CONTROL_DEMO_HARDWARE_PUBLIC
+  hardware_interface::CallbackReturn on_deactivate(
+    const rclcpp_lifecycle::State & previous_state) override;
+
+  ROS2_CONTROL_DEMO_HARDWARE_PUBLIC
+  hardware_interface::return_type read(
+    const rclcpp::Time & time, const rclcpp::Duration & period) override;
+
+  ROS2_CONTROL_DEMO_HARDWARE_PUBLIC
+  hardware_interface::return_type write(
+    const rclcpp::Time & time, const rclcpp::Duration & period) override;
+
+private:
+  std::unique_ptr<rclcpp::Logger> logger_;
+  std::unique_ptr<rclcpp::Clock> clock_;
+
+  // parameters for the RRBot simulation
+  double actuator_slowdown_;
+
+  // transmissions
+  std::vector<std::shared_ptr<transmission_interface::Transmission>> transmissions_;
+
+  struct InterfaceData
+  {
+    explicit InterfaceData(const std::string & name);
+
+    std::string name_;
+    double command_;
+    double state_;
+    double transmission_;
```

Comment:
This name is quite confusing. Could you rename this variable please to clarify it's meaning/content?
What does it mean in the joint and in the actuator context?

Reply:
It's the sink of the Joint/Actuator handle that lives inside the transmission. `transmission_passthrough_` maybe?

Reply:
If you rename plus add this comment about the sink we are done ;)